src/mapping/nav2_map_server/src/map_server.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_map_server/map_server.hpp"
+#include "nav2_map_server/map_loader.hpp"
+#include <string>
+
+#define USAGE    "\nUSAGE: map_server <map.yaml> <map_type>\n" \
+  "  map.yaml: map description file\n" \
+  "  map_type: the type of map to load (i.e. occupancy)\n"
+
+
+MappingServerROS::MappingServerROS(const std::string & fname, const std::string & map_type)
+{
+  n = rclcpp::Node::make_shared("map_server");
+
+  try {
+    m = new MapLoader();
+
+    RCLCPP_INFO(n->get_logger(), "Loading Map of Type '%s'", map_type.c_str());
+    MyMap = m->createMap(map_type, n, fname);
+    rclcpp::spin(n);
+  } catch (std::runtime_error e) {
+    RCLCPP_ERROR(n->get_logger(), "Cannot load map");
+    exit(-1);
+  }
+}
+
+
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
+
+  if (argc != 3 && argc != 2) {
+    fprintf(stderr, "[ERROR] [map_server]: %s", USAGE);
```

Comment:
This brings up an issue I had with the RCLCPP_* statements. Unlike in ROS1, they require a logger from a node, but in this case, I have yet to instantiate a node. In other cases (see function LoadMapInfoFromFile), I don't necessarily want to require a node because a user may want to instantiate the object without a node to take advantage of the map loading functionalities (as in the case of my unit tests where the node isn't needed). I could refactor to make most statements to be RCLCPP console outputs, but I think we should discuss what the standard should be for system outputs that aren't using nodes. Looking at some of the ros2 demos, apparently they just use printf outside of their node.

Reply:
Actually you don't need a node. I originally thought the same thing when I ran into this on my port of map_server internally, but the logger has a "friend" statement which allows you to call get_logger directly.  http://docs.ros2.org/latest/api/rclcpp/logger_8hpp_source.html
so you can simply use:
ROS_ERROR(rclcpp::get_logger("map_server", "%s", USAGE)